server/src/test/java/org/apache/iotdb/db/auth/AuthorityCheckerTest.java

```diff
@@ -0,0 +1,133 @@
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.DROP_INDEX, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.UNION, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), new ArrayList<>(),
+        OperatorType.INSERT, user.getName()));
+
+    Assert.assertTrue(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.MODIFY_PASSWORD, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.GRANT_USER_PRIVILEGE, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.GRANT_ROLE_PRIVILEGE, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.REVOKE_USER_PRIVILEGE, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.REVOKE_ROLE_PRIVILEGE, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.REVOKE_ROLE_PRIVILEGE, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.GRANT_USER_ROLE, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.DELETE_USER, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.DELETE_ROLE, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.LIST_ROLE, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.LIST_USER, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.SET_STORAGE_GROUP, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.CREATE_TIMESERIES, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.DELETE_TIMESERIES, user.getName()));
+  }
+}
```

Comment:
It's not a good test. The user created has no privilege at all, so all assertions return false. The correct test method may be: 1. test the authority before granting, return false. 2. grant the privilege 3. test the authority again, return true.

Reply:
update